packages/api-specification/api/windows.ts

```diff
@@ -0,0 +1,313 @@
+     * @returns {Promise<void>} A promise which resolves to nothing when the function has completed.
+     */
+    flashFrame(flag: boolean): Promise<void>;
+
+    /**
+     * Activates the window. This will bring the window to front or restore it if minimized.
+     * If focus flag is true (default) the window will also receive input focus.
+     * @returns {Promise<void>} A promise which resolves to nothing when the function has completed.
+     */
+    activate(focus?: boolean): Promise<void>;
+
+    /**
+     * Captures an image of the window
+     * @param options
+     */
+    capture(options?: CaptureOptions): Promise<Base64ImageData>;
+
+    /** Get the bounds of the window */
+    getBounds(): Bounds;
+
+    /**
+     * Get the child windows of the window.
+     */
+    getChildWindows(): Window[];
+
+    /**
+     * Get the parent of the window. Null will be returned if the window has no parent.
+     */
+    getParentWindow(): Window;
+
+    /**
+     * Returns the current state of the window - normal, maximized or minimized
+     */
+    getState(): "normal" | "maximized" | "minimized";
+
+    /**
+     * Hides the window.
+     * @returns {Promise<void>} A promise that resolves to nothing when the window has hidden.
+     */
+    hide(): Promise<void>;
+
+    /**
+     * Load a new URL in the window.
+     * @param {string} url - The URL to load in the window.
+     * @returns {Promise<void>} A promise that resolves when the window method succeeds.
+     */
+    loadURL(url: string): Promise<void>;
+
+    /**
+     * Reload the window.
+     * @returns {Promise<void>} A promise that resolves when the window method succeeds.
+     */
+    reload(): Promise<void>;
+
+    /**
+     * Restores the window to normal state.
+     * @returns {Promise<void>} A promise that resolves to nothing when the window method succeeds.
+     */
+    restore(): Promise<void>;
+
+    /** Set new bounds */
+    setBounds(bounds: Bounds);
+
+    /**
+     * Sets the window icon.
+     * @param {string} icon - The url to the image.
+     * @returns {Promise<void>} A promise that resolves to nothing when the option has been set.
+     */
+    setIcon(icon: string): Promise<void>;
+
+    /**
+     * Show the window.
+     * @returns {Promise<void>} A promise that resolves to nothing when the window is showing.
+     */
+    show(): Promise<void>;
+
+    /**
+     * Maximize the window.
+     * @returns {Promise<void>} A promise that resolves to nothing when the window has maximized.
+     */
+    maximize(): Promise<void>;
+
+    /**
+     * Minimize the window.
+     * @returns {Promise<void>} A promise that resolves to nothing when the window has minimized.
+     */
+    minimize(): Promise<void>;
+
+    /**
+     * Send a message to the window.
+     * @param {string|object} message - The message to send to the window. Can be any serializable object.
+     */
+    sendMessage(message: string | object): Promise<void>;

```

Comment:
The HTML5 API is [`postMessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) so we should maybe use that?